### PR TITLE
xrootd: define GSI security protocol params by default

### PR DESCRIPTION
**Pull request by jouvin**
This PR doesn't produce any configuration change by itself, without a new version of ncm-xrootd (https://github.com/quattor/configuration-modules-grid/pull/105) planned for 16.6 but is a requirement for deploying this new version. Because it is a requirement I think **it should be part of 16.4**.

Contributes to https://github.com/quattor/configuration-modules-grid/issues/104

### CODE REVIEW

File: personality/xrootd/env.pan [hunk 1]
```diff
@@ -35,3 +38,19 @@ variable XROOTD_MONITORING_DESTINATIONS ?= undef;
...
+  SELF["key"] = "/etc/grid-security/dpmmgr/dpmkey.pem";
+  SELF["md"] = "sha256:sha1";
+  if ( VOMS_XROOTD_EXTENSION_ENABLED ) { 
+    SELF["vomsfun"] = "/usr/lib64/libXrdSecgsiVOMS.so";
+  } else {
+    SELF["vomsat"] = 0;
+  };
+
+  SELF;
+};
```
Comment (jrha) [inline personality/xrootd/env.pan]: I find these blocks of DML that use SELF to build dicts hard to follow, what is your opinion of the following equivalent?

```pan
variable XROOTD_SECURITY_GSI_PARAMS ?= dict(
  "ca", 2,
  "cert", "/etc/grid-security/dpmmgr/dpmcert.pem",
  "crl", 3,
  "gmapopt", 10,
  "key", "/etc/grid-security/dpmmgr/dpmkey.pem",
  "md", "sha256:sha1",
  if (VOMS_XROOTD_EXTENSION_ENABLED) "vomsfun" else "vomsat",
  if (VOMS_XROOTD_EXTENSION_ENABLED) "/usr/lib64/libXrdSecgsiVOMS.so" else 0,
);
```

Comment (jouvin) [inline personality/xrootd/env.pan]: You are right! I'll try to update it later today... @Pansanel or @guillaume-philippon could you confirm that you don't see any problem merging it in 16.4. I would be important to allow a smooth deployment of https://github.com/quattor/configuration-modules-grid/pull/105 later.

**Comment by jouvin** [comment]
@Pansanel or @guillaume-philippon could you confirm that you don't see any problem merging it in 16.4. I would be important to allow a smooth deployment of https://github.com/quattor/configuration-modules-grid/pull/105 later. @guillaume-philippon is probably biased, would be better if @Pansanel could have a look!

File: personality/xrootd/env.pan (hunk 1 above)
Comment (jouvin) [inline personality/xrootd/env.pan]: @jrha I looked at your proposal more closely and I don't buy it. I agree that all these `SELF` are not as readable as your version but the 2 last lines with the `if.. else...` block will be the source of future problems. We look the `key/value` structure (at least it is not easy to identify) and at some point there is the risk of somebody changing one line without changing the other. I think my version is more maintainable.... Is it acceptable?

**Comment by Pansanel** [comment]
It looks fine for me, but I could not test it at IPHC.

**Comment by jouvin** [comment]
I am not sure if I told it... I deployed it at LAL without any impact on existing system (no resulting configuration change despite the additional info in the profie), which is the goal of this PR as explained in the description.

**Comment by jouvin** [comment]
Merging it as nobody seems to have any opposition!!
